Common/Util/ObjectActivator.cs

```diff
@@ -138,28 +124,16 @@ public static object Clone(object instanceToClone)
         }
 
         /// <summary>
-        /// Creates an activator for PythonData type
+        /// Adds method to return an instance of object
         /// </summary>
-        /// <param name="module">The algorithm python module</param>
-        public static void SetPythonModule(PyObject module)
+        /// <param name="key">The key of the method to add</param>
+        /// <param name="value">The value of the method to add</param>
+        public static void AddActivator(Type key, Func<object[], object> value)
         {
-            _activatorsByType[typeof(PythonData)] = type =>
+            if (!_activatorsByType.ContainsKey(key))
```

Comment:
I like this pattern much more tan the set pattern as it's more explicit. Great change.

Reply:
Also, what should the behavior be if `AddActivator` is called twice for the same `Type`? Should we throw? I don't think silently doing nothing is the correct approach.

Reply:
In python algorithms, `AddActivator` cannot be called twice for the same type since we check for it in QCAlgorithm.CreateType.

Reply:
This is a public static method, anyone can call it again, but shouldn't. Can we add an `else` block that throws as it's a common pattern when adding something by key and the key already exists to throw an exception, in addition, since we don't expect this case to ever happen it would be nice if it did accidentally happen in the future that we're warned about it immediately via the exception.